hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -256,6 +260,7 @@ protected void clearStorage(boolean onShutdown) {
         } else {
             storage.clear(false);
         }
+        mutationObserver.onClear();
```

Comment:
The `clearStorage` method is called from two places:
- `DefaultRecordStore#destroyInternals`. That method also calls `RecordStoreMutationObserver#onClear` which then makes it being called twice. Or if we change the invocation in `destroyInternals`, then we have two different invocations of the observer
- `DefaultRecordStore#clearPartition`. This one in turn is called from:
  - `DefaultRecordStore#destroy`, where we already have an invocation of the `observer`, so it's again called twice
  - `MapServiceContextImpl#clearMapsHavingLesserBackupCountThan` and `MapServiceContextImpl#clearPartitionData` which is ok since we don't have the observer invocation in this path yet
  - `PartitionContainer#clear` which is called from `MapServiceContextImpl#reset` when resetting the map service  and `MapServiceContext#shutdown` which is called when shutting down the map service. This path may also be ok.

So I think the problems we are facing are just that we are invoking the observer twice in these cases:
- `DefaultRecordStore#destroyInternals` -> `clearStorage`
- `DefaultRecordStore#destroy` -> `DefaultRecordStore#clearPartition` -> `clearStorage`

This may need to be fixed. OTOH, if `onClear` is idempotent and releases resources, we may leave it being called twice. This has just an impact on non-idempotent observers.